sdk/go/postgres/pg_types.go

```diff
@@ -0,0 +1,119 @@
+// generated by wit-bindgen 0.3.0
+package postgres
+
+import "fmt"
+
+type PgErrorKind int
+
+const (
+	PgErrorKindSuccess PgErrorKind = iota
+	PgErrorKindConnectionFailed
+	PgErrorKindBadParameter
+	PgErrorKindQueryFailed
+	PgErrorKindValueConversionFailed
+	PgErrorKindOtherError
+)
+
+func (k PgErrorKind) String() string {
+	return [...]string{
+		"PgErrorKindSuccess",
+		"PgErrorKindConnectionFailed",
+		"PgErrorKindBadParameter",
+		"PgErrorKindQueryFailed",
+		"PgErrorKindValueConversionFailed",
+		"PgErrorKindOtherError",
+	}[k]
+}
+
+type PgError struct {
+	kind PgErrorKind
+	val  interface{}
+}
+
+func (n PgError) Kind() PgErrorKind {
+	return n.kind
+}
+
+func PgErrorSuccess() PgError {
+	return PgError{kind: PgErrorKindSuccess}
+}
+
+func PgErrorConnectionFailed(v string) PgError {
+	return PgError{kind: PgErrorKindConnectionFailed, val: v}
+}
+
+func (n PgError) GetConnectionFailed() string {
+	if g, w := n.Kind(), PgErrorKindConnectionFailed; g != w {
+		panic(fmt.Sprintf("Attr kind is %v, not %v", g.String(), w.String()))
```

Comment:
Again panics seem quite harsh, especially given that the developer can't expect errors the way they can data types... is the expectation that the app would call `Kind()` before calling the relevant Get function?